unittests/allplatformstests.py

```diff
@@ -1901,6 +1901,25 @@ def test_options_with_choices_changing(self) -> None:
                 self.assertEqual(item['value'], ['b', 'c'])
                 self.assertEqual(item['choices'], ['b', 'c', 'd'])
 
+    def test_options_listed_in_build_options(self) -> None:
+        """Detect when changed options become listed in build options."""
+        testdir = Path(os.path.join(self.unit_test_dir, '110 list build options'))
+
+        out = self.init(str(testdir))
+        for line in out.splitlines():
+            if line.startswith('Message: Build options:'):
+                self.assertNotRegex(line, '-Dauto_features=auto')
+                self.assertNotRegex(line, '-Doptional=auto')
+
+        self.wipe()
+        self.mac_ci_delay()
+
+        out = self.init(str(testdir), extra_args=['-Dauto_features=disabled', '-Doptional=enabled'])
+        for line in out.splitlines():
+            if line.startswith('Message: Build options:'):
+                self.assertRegex(line, '-Dauto_features=disabled')
+                self.assertRegex(line, '-Doptional=enabled')
```

Comment:
None of these tests are regular expressions so it should not use `assertRegex` but `assertIn`.

This test should also test what happens when you something like this:

- set up project
- check value of build options
- change some option with `meson configure`
- check value of build options again

Should the values of the two checks be the same or different? What values exactly?

Reply:
I've just added the requested checks to the test.

Reply:
This doesn't test the right thing... it checks if freshly initing changes the value. Per @jpakkane's review, we want to check what happens when you run `self.setconf()` with some value to be set.

`self.build()` should then return output that includes the contents of a reconfigure.

Reply:
Also -- this still contains assertRegex.